### DISCUSSION TITLE
How to change the username on mastodon

### DISCUSSION THREAD

**Discussion by gamersalpha** [mastodon/mastodon#19846] [Q&A]
Hello evrybody, so in stlal my own mastodon, and i start to follow pepole from other mastodon taht's so cool !!

so now i got an problem with my username, i need to change it ?

i can't found where to change it ?

so i think to option : 

1/ i remove my acocunt from scratch and i change the username, but i will lost all my follow and all my post ?
2/ maybe some where i can change the username of my account

how do that please

**Comment by trwnh** [comment]
It's like email. You can't change your username because it's part of your address, and all links to your post would break because they currently depend on your username. See #17030 if you are curious about the technical changes that would be necessary before this would be possible.

**Comment by gamersalpha** [comment]
ok i got it

thanks for u response, i have to change remove evrything

thanks again

**Comment by cclambie** [comment]
You can create a new account, with the username you want, then "take" all your followers from your old account to the new one.

With the export/import function https://docs.joinmastodon.org/user/moving/
I can't see why this wouldn't work on the same server instance.
